Week2/js/HomeTask2.js

```diff
@@ -0,0 +1,204 @@
+document.title = "FrontCamp Training Task Two";
+var mainDivision = document.createElement("div");
+var dataDiv = document.createElement("div");
+function CreatePage() {
+  // navbar
+  let header = document.createElement("div");
+  header.className = "header";
+  header.setAttribute("class", "header");
+  let title = document.createElement("h1");
+  title.innerText = "NEWSFEED";
+  title.setAttribute("class", "newsfeed");
+  let another_feedback = document.createElement("h4");
+  another_feedback.className = "another-feedback";
+  another_feedback.innerText = "Yet another feedback";
+  header.appendChild(title);
+  header.appendChild(another_feedback);
+  document.body.appendChild(header);
+  // navbar end
+  CreateContent();
+}
+
+// this method creates the main content of the body
+var CreateContent = function() {
+  mainDivision.setAttribute("class", "maindiv");
+  mainDivision.setAttribute("id", "maindiv");
+
+  let mainContainer = document.createElement("div");
+  mainContainer.setAttribute("class", "flex-container");
+  let logo = document.createElement("div");
+  logo.setAttribute("class", "logo");
+  let content = document.createElement("div");
+  content.setAttribute("class", "content");
+  mainContainer.appendChild(logo);
+  mainContainer.appendChild(content);
+
+  var category = document.createElement("div");
+  category.setAttribute("class", "category");
+  var categories = createCategories();
+  var emailInputField = document.createElement("input");
+  emailInputField.setAttribute("type", "text");
+  emailInputField.setAttribute("name", "fname");
+  emailInputField.setAttribute("class", "input");
+  emailInputField.setAttribute("id", "email");
+  emailInputField.addEventListener("blur", validateEmailId);
+  var subscribe = document.createElement("button");
+  subscribe.setAttribute("value", "Subscribe");
+  subscribe.setAttribute("class", "subscribe-button");
+  subscribe.innerText = "Subscribe";
+  category.appendChild(categories);
+  category.appendChild(emailInputField);
+  category.appendChild(subscribe);
+  mainContainer.appendChild(category);
+  mainDivision.appendChild(mainContainer);
+  dataDiv.setAttribute("id", "data-div");
+  displayAll(dataDiv);
+  document.body.appendChild(mainDivision);
+};
+var generateItems = function() {
+  let divisions = [];
+  const titleNames = [
+    "Canada",
+    "Mexico",
+    "Us",
+    "Austria",
+    "Bulgaria",
+    "Republic",
+    "Germany",
+    "Hungary",
+    "Ireland",
+    "Netherlands"
+  ];
+  let logoPaths = [
+    "./Assets/canada.png",
+    "./Assets/mexico.png",
+    "./Assets/us.png",
+    "./Assets/austria.png",
+    "./Assets/bulgaria.png",
+    "./Assets/republic.png",
+    "./Assets/germany.png",
+    "./Assets/hungary.png",
+    "./Assets/ireland.png",
+    "./Assets/netherlands.png"
+  ];
+  // this loop will create 10 items
+  for (var i = 0; i < 10; i++) {
+    let mainContainer = document.createElement("div");

```

Comment:
You can rename it to tile or news-feed for a more meaningful name than mainContainer. It gives assumption that you are creating mainContainer again.